### PR TITLE
Replace freemarker with stringsubstitutor from commons-text for template expanding

### PR DESCRIPTION
**Pull request by Aloren**
Freemarker is very powerful template engine, but in Kayenta only 1 percent of it's capabilities are actually used. For template expanding with map of variables simple string substitutor does the trick.
Also this change allowed to remove `throws IOException` from the signature of `MetricsService.buildQuery`.

Added tests for all basic scenarios of `QueryConfigUtils`.

### CODE REVIEW

**Comment by Aloren** [comment]
Also regarding the issue with the newly added test `expandCustomFilter_returnsExpandedTemplateWithExtractedBaseScopeAttributesFromFilterTemplate`. If baseScopeAttribute contains properties both from `CanaryConfig` and `CanaryMetricSetQueryConfig` ( like "scope", "location", "serviceType" ), `populateTemplateBindings` will fail with exception `java.lang.IllegalArgumentException: Unable to find property 'serviceType'.` Looks like this is a bug in `populateTemplateBindings`. If you don't mind I would like to address that issue in a separate pull request.

File: kayenta-stackdriver/src/test/groovy/com/netflix/kayenta/providers/metrics/StackdriverCanaryMetricSetQueryConfigSpec.groovy [hunk 1]
```diff
@@ -92,7 +92,6 @@ class StackdriverCanaryMetricSetQueryConfigSpec extends Specification {
     ["my-template-1": 'A test: key1=${key1}.',
      "my-template-2": 'A test: key2=${key2}.']                | "my-template-x"        | null
     [:]                                                       | "my-template-x"        | null
-    null                                                      | "my-template-x"        | null
```
Comment (Aloren) [inline kayenta-stackdriver/src/test/groovy/com/netflix/kayenta/providers/metrics/StackdriverCanaryMetricSetQueryConfigSpec.groovy]: Added `@Singular` for `templates` field in `CanaryConfig`, so it is not valid to use: `CanaryConfig.builder().templates(null).build()`, that's why I deleted this case.

**Comment by fieldju** [comment]
I am going to defer review of this to @duftler as he made the original template integration and could speak to this change better than I.

Out of curiosity what is the problem this PR solves?

**Comment by Aloren** [comment]
I was working on controller tests and saw that the context was starting too slow, digged into the context and found freemarker in the dependencies. Initially thought that this was added by mistake, but then found that this was used for replacing placeholders in string. I personally think that this is excessive and since the change was trivial I submitted PR. (Also it is possible to use standard spel for such things as an alternative)

File: kayenta-core/src/test/groovy/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020 Netflix, Inc.
```
Comment (duftler) [inline kayenta-core/src/test/groovy/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtilsTest.java]:
```suggestion
 * Copyright 2020 Playtika, Inc.
```

File: kayenta-core/src/test/groovy/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtilsSpec.groovy [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /*
- * Copyright 2018 Google, Inc.
+ * Copyright 2020 Netflix, Inc.
```
Comment (duftler) [inline kayenta-core/src/test/groovy/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtilsSpec.groovy]:
```suggestion
 * Copyright 2018 Google, Inc.
```

File: kayenta-core/src/test/groovy/com/netflix/kayenta/canary/providers/metrics/TestCanaryMetricSetQueryConfig.groovy [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 Netflix, Inc.
```
Comment (duftler) [inline kayenta-core/src/test/groovy/com/netflix/kayenta/canary/providers/metrics/TestCanaryMetricSetQueryConfig.groovy]:
```suggestion
 * Copyright 2020 Playtika, Inc.
```

**Comment by duftler** [comment]
One other minor thing, I think the commit message should be: "refactor(core): Replace freemarker..." instead of "feat(core): Replace freemarker...".

File: kayenta-core/src/main/java/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtils.java [hunk 5]
```diff
@@ -51,8 +46,7 @@ public static String expandCustomFilter(
       CanaryConfig canaryConfig,
       CanaryMetricSetQueryConfig metricSetQuery,
       CanaryScope canaryScope,
-      String[] baseScopeAttributes)
-      throws IOException {
```
Comment (nisanharamati) [inline kayenta-core/src/main/java/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtils.java]: Nice!

**Comment by Aloren** [comment]
Absolutely love the part when you need to fix copyright headers.. :D

**Comment by Aloren** [comment]
> Once the copyright messages are fixed and the commit message is updated, all that remains is for someone from Netflix to have a look.

Just fixed.
Thank you for the review!

**Pull request merged**